### PR TITLE
OBAM-144 Improvement to forecast report

### PR DESCRIPTION
**Pull request by MagdaToczek**

### CODE REVIEW

File: grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy [hunk 1]
```diff
@@ -688,7 +688,7 @@ class ReportController {
                 }
                 def replenishmentPeriodDays = params.replenishmentPeriodDays ?: inventoryLevel && inventoryLevel.replenishmentPeriodDays ? inventoryLevel.replenishmentPeriodDays : 365
                 def leadTimeDays = params.leadTimeDays ?: inventoryLevel && inventoryLevel.expectedLeadTimeDays ? inventoryLevel.expectedLeadTimeDays : 365
-                def productExpiry = forecastingService.getProductExpiry(Location.load(params.originId), replenishmentPeriodDays + leadTimeDays, item.productId)
+                def productExpiry = forecastingService.getProductExpiry(Location.load(params.originId), replenishmentPeriodDays.toInteger() + leadTimeDays.toInteger(), item.productId)
```
Comment (jmiranda) [inline grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy:691]: @MagdaToczek Could you explain the need for this?

File: grails-app/services/org/pih/warehouse/report/ReportService.groovy [hunk 2]
```diff
@@ -688,12 +692,36 @@ class ReportService implements ApplicationContextAware {
         return data
     }
 
+    Map getQtyOnOrder (String locationId) {
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/report/ReportService.groovy:695]: Let's spell this out ... so getQuantityOnOrder. Same for variables below and wherever possible.

**Review by jmiranda — CHANGES_REQUESTED** [review]

File: grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy (hunk 1 above)
Comment (MagdaToczek) [reply grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy:691]: Sure, in previous version it was concatenation of strings... And that wasn't my intention :)

File: grails-app/services/org/pih/warehouse/report/ReportService.groovy (hunk 2 above)
Comment (MagdaToczek) [reply grails-app/services/org/pih/warehouse/report/ReportService.groovy:695]: ok

File: grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy (hunk 1 above)
Comment (MagdaToczek) [reply grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy]: I corrected this a bit and change to direct typing of this two variables.

File: grails-app/services/org/pih/warehouse/report/ReportService.groovy [hunk 3]
```diff
@@ -688,12 +692,36 @@ class ReportService implements ApplicationContextAware {
         return data
     }
 
+    Map getQuantityOnOrder (String locationId) {
+        def location = Location.get(locationId)
+        def items = orderService.getPendingInboundOrderItems(location)
```
Comment (awalkowiak) [inline grails-app/services/org/pih/warehouse/report/ReportService.groovy:697]: Add an option to `getPendingInboundOrderItems` by location and list of products (there is already by location and product)

File: grails-app/services/org/pih/warehouse/report/ReportService.groovy [hunk 4]
```diff
@@ -688,12 +692,36 @@ class ReportService implements ApplicationContextAware {
...
+        items += shipmentService.getPendingInboundShipmentItems(location)
```
Comment (awalkowiak) [inline grails-app/services/org/pih/warehouse/report/ReportService.groovy:698]: Same as above

**Review by awalkowiak — CHANGES_REQUESTED** [review]

File: grails-app/services/org/pih/warehouse/report/ReportService.groovy (hunk 3 above)
Comment (MagdaToczek) [reply grails-app/services/org/pih/warehouse/report/ReportService.groovy]: Ok, done

File: grails-app/services/org/pih/warehouse/report/ReportService.groovy (hunk 4 above)
Comment (MagdaToczek) [reply grails-app/services/org/pih/warehouse/report/ReportService.groovy]: Ok, done

**Review by awalkowiak — APPROVED** [review]

File: grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy (hunk 1 above)
Comment (jmiranda) [reply grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy]: Ah ok that makes sense.

**Pull request merged**